marklogic-data-hub/src/main/java/com/marklogic/hub/HubProject.java

```diff
@@ -247,7 +248,7 @@ public interface HubProject {
      *
      * @throws IOException if problem happens with the on-disk project.
      */
-    void upgradeProject(FlowManager flowManager) throws IOException;
+    void upgradeProject(FlowManagerImpl flowManager) throws IOException;
```

Comment:
Any reason we are "downgrading" to the impl class?
I believe the correct way should be to cast it on the implementation code of updateProject.

Reply:
The addStepToFlow is not a user-facing method and is not exposed in the interface